src/Inkluzitron/Handlers/InviteHandler.cs

```diff
@@ -0,0 +1,74 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Discord.WebSocket;
+using Inkluzitron.Data;
+using Inkluzitron.Extensions;
+using Inkluzitron.Services;
+using Microsoft.EntityFrameworkCore;
+
+namespace Inkluzitron.Handlers
+{
+    /// <summary>
+    /// Handler to catch event on joining new user
+    /// </summary>
+    public class InviteHandler : IHandler
+    {
+        private DiscordSocketClient DiscordSocketClient { get; }
+
+        private DatabaseFactory DatabaseFactory { get; }
+        private UsersService UsersService { get; }
+
+        public InviteHandler(DiscordSocketClient discordSocketClient,
+            DatabaseFactory databaseFactory,
+            UsersService usersService)
+        {
+            DiscordSocketClient = discordSocketClient;
+            DatabaseFactory = databaseFactory;
+            UsersService = usersService;
+
+            DiscordSocketClient.UserJoined += OnUserJoinedAsync;
+        }
+
+        /// <summary>
+        /// Checks which invite link has been used by newly joined user and
+        /// updates corresponding record in database for it
+        /// </summary>
+        /// <param name="user">newly joined user</param>
+        private async Task OnUserJoinedAsync(SocketGuildUser user)
+        {
+            var dbContext = DatabaseFactory.Create();
+            var inviteLinks = await user.Guild.GetInvitesAsync();
+            var inviteeDb = await dbContext.GetOrCreateUserEntityAsync(user);
+
+            foreach (var link in inviteLinks)
+            {
+                if (link.Inviter.Id != DiscordSocketClient.CurrentUser.Id) continue;
+                if (link.Uses == 0) continue;
+
+                var invite = await dbContext.Invites.AsQueryable()
+                    .Where(i => i.InviteLink == link.Url)
+                    .FirstAsync();
```

Comment:
```suggestion
                    .FirstOrDefaultAsync();
```

Tady POZOR! FirstAsync vyhodí výjimku, pokud nenalezne žádný prvek. Pokud chceš takové invites ignorovat (v reakci na if na řádku 52), pak musíš použít něco, co ti vrátí výchozí hodnotu dané třídy (neboli null).